### ISSUE TITLE
federation: deleting a service does not delete all DNS records

### ISSUE THREAD

**Issue by nikhiljindal** [kubernetes/kubernetes#36799]
Steps to repro:
* Create a federated service and then delete it.

Expected: All DNS records created for that federated service should go away.
Actual: CNAME records still remain.

Details:
Running `
$ gcloud dns record-sets list --zone=nikhiljindal` after deleting federated service gives me
```
nginx.default.federation.svc.us-central1.nikhiljindal.test-f8n.k8s.io.                CNAME  180    nginx.default.federation.svc.nikhiljindal.test-f8n.k8s.io.
nginx.default.federation.svc.us-central1-a.us-central1.nikhiljindal.test-f8n.k8s.io.  CNAME  180    nginx.default.federation.svc.us-central1.nikhiljindal.test-f8n.k8s.io.
nginx.default.federation.svc.us-central1-f.us-central1.nikhiljindal.test-f8n.k8s.io.  CNAME  180    nginx.default.federation.svc.us-central1.nikhiljindal.test-f8n.k8s.io.
```
I had created a service called nginx and have 2 registered clusters one in us-central1-a and other in us-central1-f.

I see that there is no e2e test for this, so not sure if we broke this recently or its always been like this.

cc @kubernetes/sig-cluster-federation @mfanjie

**Comment by nikhiljindal** [comment]
@mfanjie Can you work on this?

**Comment by quinton-hoole** [comment]
@nikhiljindal I'm not sure that you want this behavior by default, as it will cause all of the clients using the service via its external DNS names to experience a global outage, which I can imagine would not be what they would want or expect. I would suggest that this be a non-default delete-option, along the lines of cascading deletion.

**Comment by nikhiljindal** [comment]
Ah you are right. I got confused because it deleted some records (all A records) but not all (not the CNAME records).
It should not remove any DNS records at all.

@madhusudancs pointed out that it might be removing A records because it might be deleting the services from underlying clusters when federated service is deleted.
I verified and it is indeed the case.

So the fix here is that we should not delete services from underlying clusters (unless cascading deletion is explicitly requested by user). And then we can let all DNS records be as is.
